src/Aspire.Hosting/Publishing/PublishingActivityProgressReporter.cs

```diff
@@ -116,6 +134,29 @@ internal PublishingTask(string id, string stepId, string statusText)
     /// The progress reporter that created this task.
     /// </summary>
     internal IPublishingActivityProgressReporter? Reporter { get; set; }
+
+    /// <summary>
+    /// Completes the task automatically when disposed if not already completed.
+    /// </summary>
+    public void Dispose()
```

Comment:
This should be IAsyncDisposable, and it should not throw (we can remove these catch methods)